engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/StraightThroughProcessingLoopValidationTest.java

```diff
@@ -0,0 +1,231 @@
+    // given
+    final var processId = Strings.newRandomValidBpmnId();
+
+    // when
+    final var rejectedDeployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .task("task1")
+                    .task("task2")
+                    .task("task3")
+                    .task("task4")
+                    .task("task5")
+                    .task("task6")
+                    .task("task7")
+                    .task("task8")
+                    .task("task9")
+                    .connectTo("task7")
+                    .done())
+            .expectRejection()
+            .deploy();
+
+    // then
+    Assertions.assertThat(rejectedDeployment)
+        .hasKey(ExecuteCommandResponseDecoder.keyNullValue())
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+    assertThat(rejectedDeployment.getRejectionReason())
+        .contains(String.format("Process: %s", processId))
+        .contains(GENERIC_REJECTION_MESSAGE + "task7 > task8 > task9 > task7");
+  }
+
+  @Test
+  public void shouldRejectDeploymentWithCollaborationContainingStraightThroughProcessingLoop() {
+    // given
+    final String resource = "/processes/collaboration_straight_through_processing_loop.bpmn";
+
+    // when
+    final Record<DeploymentRecordValue> rejectedDeployment =
+        ENGINE.deployment().withXmlClasspathResource(resource).expectRejection().deploy();
+
+    // then
+    Assertions.assertThat(rejectedDeployment)
+        .hasKey(ExecuteCommandResponseDecoder.keyNullValue())
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+    assertThat(rejectedDeployment.getRejectionReason())
+        .contains("Process: process1")
+        .contains(GENERIC_REJECTION_MESSAGE + "task1 > task2 > task1")
+        .contains("Process: process2")
+        .contains(GENERIC_REJECTION_MESSAGE + "manualTask2 > manualTask1 > manualTask2");
+  }
+
+  @Test
+  public void shouldDeployProcessWithRegularLoops() {
+    // given
+    final var processId = Strings.newRandomValidBpmnId();
+
+    // when
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .userTask("task1")
+                    .userTask("task2")
+                    .connectTo("task1")
+                    .done())
+            .deploy();
+
+    // then
+    assertThat(deployment.getKey())
+        .describedAs("Allow deployments of loops that aren't straight-through processed")
+        .isNotNegative();
+  }
+
+  // TODO create test to verify multiple processes in 1 file
```

Comment:
❓ wanna still do that before merge?

Reply:
Thanks for raising. The test is there (`shouldRejectDeploymentWithCollaborationContainingStraightThroughProcessingLoop`), I just forgot to remove the TODO 😄